lisa/tests/staging/schedutil.py

```diff
@@ -0,0 +1,271 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2019, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from math import ceil
+import os
+
+import pandas as pd
+
+from lisa.wlgen.rta import Ramp
+from lisa.tests.base import TestBundle, ResultBundle, Result, RTATestBundle
+from lisa.target import Target
+from lisa.trace import requires_events, FtraceCollector, FtraceConf
+from lisa.datautils import df_merge, series_mean
+from lisa.utils import ArtifactPath
+
+from lisa.analysis.rta import PerfAnalysis
+from lisa.analysis.frequency import FrequencyAnalysis
+from lisa.analysis.load_tracking import LoadTrackingAnalysis
+
+
+class RampBoostTestBase(RTATestBundle):
+    """
+    Test schedutil's ramp boost feature.
+
+    """
+
+    def __init__(self, res_dir, plat_info, cpu):
+        super().__init__(res_dir, plat_info)
+        self.cpu = cpu
+
+    @requires_events('cpu_idle', 'cpu_frequency', 'sched_wakeup')
+    def estimate_nrg(self):
+        return self.plat_info['nrg-model'].estimate_from_trace(self.trace).sum(axis=1)
+
+    def get_avg_slack(self, only_negative=False):
+        pa = PerfAnalysis.from_dir(self.res_dir)
```

Comment:
I'm late to the party but this'll have to be updated to the new ftrace-based analysis :(